### PR TITLE
Allow `cugraph` to be imported in an SG env for SG algorithms

### LINKED ISSUE #2113: [BUG] importing cugraph in Amazon SageMaker Studio Lab cause import error libnuma.so.1
**Describe the bug**
importing cugraph in Amazon SageMaker Studio Lab cause import error libnuma.so.1: cannot open shared object file: No such file or directory

**Steps/Code to reproduce bug**
import cugraph in a cell within jupyter notebook

**Expected behavior**
successful import of cugraph

**Environment overview (please complete the following information)**
Followed the exact installation from the doc: 
conda create -n rapids-22.02 -c rapidsai -c nvidia -c conda-forge rapids=22.02 python=3.8 cudatoolkit=11.4 dask-sql ipykernel -y

**Environment details**
Cannot import cugraph so I can't get the message

**Additional context**
I have no issue importing other cu libraries like cudf/dask_cudf etc.

### PR DESCRIPTION
**Pull request by betochimas**
This PR aims to directly resolve #2113, and allows for cugraph to be imported without errors in environments with only a single GPU

What this PR does:

- Moves `comms` module into the `dask` module, renaming imports across `cugraph` accordingly
- Adds mock class in case ucp is not available or usable in SG-only environments (note ucp requires libnuma.so)
- Renames `tests/dask` to `tests/mg` to better reflect the fact that those tests are for mg algs and utils
- Separates symmetrize testing into `tests/mg` and `tests` for separation of SG and MG testing

### CODE REVIEW

**Comment by betochimas** [comment]
rerun tests

**Comment by betochimas** [comment]
rerun tests

File: ci/test.sh [hunk 1]
```diff
@@ -91,7 +91,8 @@ echo "Ran Python pytest for pylibcugraph : return code was: $?, test script exit
 
 echo "Python pytest for cuGraph (single-GPU only)..."
 cd ${CUGRAPH_ROOT}/python/cugraph/cugraph
-pytest --cache-clear --junitxml=${CUGRAPH_ROOT}/junit-cugraph-pytests.xml -v --cov-config=.coveragerc --cov=cugraph --cov-report=xml:${WORKSPACE}/python/cugraph/cugraph-coverage.xml --cov-report term --ignore=raft --ignore=tests/dask --benchmark-disable
+# pytest --cache-clear --junitxml=${CUGRAPH_ROOT}/junit-cugraph-pytests.xml -v --cov-config=.coveragerc --cov=cugraph --cov-report=xml:${WORKSPACE}/python/cugraph/cugraph-coverage.xml --cov-report term --ignore=raft --ignore=tests/dask --benchmark-disable
```
Comment (rlratzel) [inline ci/test.sh:94]: You can remove the commented out line now. You might also add a comment explaining why rmat is also being skipped.

File: python/cugraph/cugraph/dask/common/input_utils.py [hunk 2]
```diff
@@ -19,8 +19,18 @@
...
+# FIXME: this raft import breaks the library if ucx-py is
+# not available. They are necessary only when doing MG work.
+from cugraph.dask.common.read_utils import MissingUCXPy
+try:
+    from raft.dask.common.utils import get_client
+except ModuleNotFoundError as err:
+    if err.name == "ucp":
+        get_client = MissingUCXPy()
+    else:
+        raise ModuleNotFoundError
```
Comment (rlratzel) [inline python/cugraph/cugraph/dask/common/input_utils.py:33]: This should just be a `raise`, which re-raises the original exception.
```suggestion
        raise 
```
If you do a `raise ModuleNotFoundError`, you lose the original information (mainly the missing module name)

**Review by rlratzel — CHANGES_REQUESTED** [review]

**Review by rlratzel — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Comment by betochimas** [comment]
rerun tests

**Comment by rlratzel** [comment]
@gpucibot merge

**Pull request merged**